app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheet.kt

```diff
@@ -105,37 +77,40 @@ class ScheduleAppointmentSheet : BottomSheetActivity() {
   }
 
   private fun screenCreates(): Observable<UiEvent> {
-    val uuid = intent.extras.getSerializable(KEY_PATIENT_UUID) as UUID
+    val patientUuid = intent.extras.getSerializable(KEY_PATIENT_UUID) as UUID
     return Observable.just(ScheduleAppointmentSheetCreated(
-        defaultDateIndex = possibleDates.indexOf(oneMonth),
-        patientUuid = uuid,
-        numberOfDates = possibleDates.size
+        patientUuid = patientUuid
     ))
   }
 
-  private fun incrementClicks() = RxView.clicks(incrementDateButton).map { AppointmentDateIncremented(currentIndex, possibleDates.size) }
+  private fun incrementClicks() = RxView.clicks(incrementDateButton).map { AppointmentDateIncremented }
+
+  private fun decrementClicks() = RxView.clicks(decrementDateButton).map { AppointmentDateDecremented }
 
-  private fun decrementClicks() = RxView.clicks(decrementDateButton).map { AppointmentDateDecremented(currentIndex, possibleDates.size) }
+  private fun notNowClicks() = RxView.clicks(notNowButton).map { SchedulingSkipped }
 
-  private fun notNowClicks() = RxView.clicks(notNowButton).map { SchedulingSkipped() }
+  private fun doneClicks() = RxView.clicks(doneButton).map { AppointmentDone }
 
-  private fun doneClicks() = RxView.clicks(doneButton).map { AppointmentScheduled(possibleDates[currentIndex]) }
+  private fun appointmentDateClicks() = RxView.clicks(currentAppointmentDate).map { ManuallySelectAppointmentDateClicked }
 
   fun closeSheet() {
     setResult(Activity.RESULT_OK)
     finish()
   }
 
-  fun closeSheet(appointmentDate: LocalDate) {
-    val intent = Intent()
-    intent.putExtra(EXTRAS_SAVED_APPOINTMENT_DATE, appointmentDate)
-    setResult(Activity.RESULT_OK, intent)
-    finish()
+  fun updateScheduledAppointment(appointmentDate: LocalDate, timeToAppointment: TimeToAppointment) {
```

Comment:
Good point, I'll make this change.

Reply:
They are not exactly the same thing. The `appointmentDate` is the exact date on which the appointment will be scheduled. The `TimeToAppointment` is what to display it as. For the same `appointmentDate` can be shown in multiple `TimeToAppointment` ways (for example, `Weeks(1)` and `Days(7)` will both point to the same `appointmentDate` but the presentation will differ because of the type.

We have tests that ensure that the same value passed in the config is passed down to the screen when displaying it. We will depend on manual testing to ensure that the configuration itself is declared correctly.